### ISSUE TITLE
local scraper not working

### ISSUE THREAD

**Issue by djbijo** [tknorris/salts#84]
for me, local scraper doesn't work.

here's a sample log, you can clearly see that the local episode is found but i don't get why it's not taken
<code>
21:35:04 T:1360262064   DEBUG: Stream All The Sources: Calling get with timeout: 20
21:35:04 T:906757040   DEBUG: RunQuery took 91 ms for 1 items query: SELECT * FROM tvshowview  WHERE ((tvshowview.c00 LIKE '%Better Call
 Saul%')) AND ((CAST(tvshowview.c05 as DECIMAL(5,1)) LIKE '%2015%'))
21:35:04 T:906757040   DEBUG: Stream All The Sources: Search Meta: {u'jsonrpc': u'2.0', u'id': u'libTvShows', u'result': {u'limits': {u'
start': 0, u'total': 1, u'end': 1}, u'tvshows': [{u'tvshowid': 11, u'title': u'Better Call Saul', u'year': 2015, u'label': u'Better Call
 Saul'}]}}
21:35:04 T:906757040   DEBUG: RunQuery took 4 ms for 2 items query: select * from episodeview  WHERE ((episodeview.idShow = 11) AND ((ep
isodeview.c12 = 1 or (episodeview.c12 = 0 and (episodeview.c15 = 0 or episodeview.c15 = 1))))) AND (((CAST(episodeview.c13 as DECIMAL(5,
1)) = 7)))
21:35:04 T:906757040   DEBUG: Stream All The Sources: Episode Meta: {u'jsonrpc': u'2.0', u'id': u'libTvShows', u'result': {u'episodes':
[{u'episode': 7, u'title': u'Bingo', u'season': 1, u'episodeid': 3716, u'label': u'1x07. Bingo', u'file': u'/media/gils/tv shows-stream/
Better Call Saul/Season 1/Better.Call.Saul.S01E07.strm', u'streamdetails': {u'video': [], u'audio': [], u'subtitle': []}}, {u'episode':
7, u'title': u'Bingo', u'season': 1, u'episodeid': 3717, u'label': u'1x07. Bingo', u'file': u'rar://%2fmedia%2fgils%2fTv%20Shows%2fBette
r%20Call%20Saul%2fSeason%2001%2fBetter.Call.Saul.S01E07.720p.HDTV.X264-DIMENSION%2fbetter.call.saul.107.720p-dimension.rar/Better.Call.S
aul.S01E07.720p.HDTV.X264-DIMENSION.mkv', u'streamdetails': {u'video': [{u'stereomode': u'', u'width': 1280, u'codec': u'h264', u'aspect
': 1.7777780294418335, u'duration': 2822, u'height': 720}], u'audio': [{u'channels': 6, u'codec': u'ac3', u'language': u''}], u'subtitle
': []}}], u'limits': {u'start': 0, u'total': 2, u'end': 2}}}
21:35:04 T:906757040   DEBUG: RunQuery took 4 ms for 2 items query: select * from episodeview  WHERE ((episodeview.idShow = 11) AND ((ep
isodeview.c12 = 1 or (episodeview.c12 = 0 and (episodeview.c15 = 0 or episodeview.c15 = 1))))) AND (((episodeview.c00 LIKE 'Bingo')))
21:35:04 T:906757040   DEBUG: Stream All The Sources: Episode Title Meta: {u'jsonrpc': u'2.0', u'id': u'libTvShows', u'result': {u'episo
des': [{u'episode': 7, u'title': u'Bingo', u'season': 1, u'episodeid': 3716, u'label': u'1x07. Bingo', u'file': u'/media/gils/tv shows-s
tream/Better Call Saul/Season 1/Better.Call.Saul.S01E07.strm', u'streamdetails': {u'video': [], u'audio': [], u'subtitle': []}}, {u'epis
ode': 7, u'title': u'Bingo', u'season': 1, u'episodeid': 3717, u'label': u'1x07. Bingo', u'file': u'rar://%2fmedia%2fgils%2fTv%20Shows%2
fBetter%20Call%20Saul%2fSeason%2001%2fBetter.Call.Saul.S01E07.720p.HDTV.X264-DIMENSION%2fbetter.call.saul.107.720p-dimension.rar/Better.
Call.Saul.S01E07.720p.HDTV.X264-DIMENSION.mkv', u'streamdetails': {u'video': [{u'stereomode': u'', u'width': 1280, u'codec': u'h264', u'
aspect': 1.7777780294418335, u'duration': 2822, u'height': 720}], u'audio': [{u'channels': 6, u'codec': u'ac3', u'language': u''}], u'su
btitle': []}}], u'limits': {u'start': 0, u'total': 2, u'end': 2}}}
21:35:04 T:906757040   DEBUG: Stream All The Sources: Local returned 0 sources from <Thread(Thread-1, started daemon 906757040)>
21:35:04 T:1360262064   DEBUG: Stream All The Sources: Got 0 Source Results
21:35:04 T:1360262064  NOTICE: Stream All The Sources: All source results received
21:35:04 T:1360262064   DEBUG: Stream All The Sources: In Reap: [<Thread(Thread-1, stopped daemon 906757040)>]
21:35:04 T:1360262064   DEBUG: Stream All The Sources: Reaping: Thread-1
21:35:04 T:1360262064  NOTICE: Stream All The Sources: No Sources found for: ||Episode|Better Call Saul|2015|1|7|Bingo|2015-03-17||
21:35:04 T:1360262064   DEBUG: Stream All The Sources: In Reap: []
</code>

**Comment by tknorris** [comment]
Do you mind pasting more of the before and after of your log to pastebin and linking it here? It's hard to follow in the comment and there's not much information to go on for what could be causing the problem in just those few lines.

**Comment by tknorris** [comment]
I see the problem. The problem is you have both a strm file for this episode and an mkv for it. Right now SALTS only pulls the first result from the library which in your case is the strm file and strm files aren't allowed in the local scraper. I'll look to see if I can fix it in some way in the next build.

**Comment by tknorris** [comment]
Fix commited. IT will show up in the next build.

**Comment by djbijo** [comment]
wow 10x for the quick fix
